test/validation/api/ipsec/ipsec.c

```diff
@@ -973,6 +973,21 @@ int ipsec_suite_init(void)
 	return rc < 0 ? -1 : 0;
 }
 
+void ipsec_prep_part_from_pkt(ipsec_test_part *part,
+			      odp_packet_t *pkt,
+			      ipsec_test_packet *pkt_in)
```

Comment:
This only sets the pkt_in field and leaves 'part' otherwise uninitialized, which is a bit unexpected. The calling function, ipsec_out_verify_send(), also leaves the structure otherwise uninitialized but passes it to ipsec_send_out(). Luckily, ipsec_send_out() does not care about the other fields that pkt_in. This is messy and fragile.

It would be cleaner if this function prepared ipsec_test_packet, not ipsec_test_part, and if ipsec_send_out() took ipsec_test_packet, not ipsec_test_part, as a parameter.

Reply:
Agreed. Will have this addressed in next version.